luigi/tools/range.py

```diff
@@ -645,6 +647,76 @@ def infer_bulk_complete_from_fs(datetimes, datetime_to_task, datetime_to_re):
     return missing_datetimes
 
 
+class RangeMonthly(RangeBase):
+    """
+    Produces a contiguous completed range of a monthly recurring task.
+
+    Unlike the Range* classes with shorter intervals, this class does not perform bulk optimisation.
+    It is aassumed that the number of months is low enough not to motivate the increased complexity.
+    Hence, there is no class RangeMonthlyBase.
+    """
+    start = luigi.MonthParameter(
+        default=None,
+        description="beginning month, inclusive. Default: None - work backward forever (requires reverse=True)")
+    stop = luigi.MonthParameter(
+        default=None,
+        description="ending month, exclusive. Default: None - work forward forever")
+    months_back = luigi.IntParameter(
+        default=13,  # Little over a year
+        description=("extent to which contiguousness is to be assured into "
+                     "past, in months from current time. Prevents infinite loop "
+                     "when start is none. If the dataset has limited retention"
+                     " (i.e. old outputs get removed), this should be set "
+                     "shorter to that, too, to prevent the oldest outputs "
+                     "flapping. Increase freely if you intend to process old "
+                     "dates - worker's memory is the limit"))
+    months_forward = luigi.IntParameter(
+        default=0,
+        description="extent to which contiguousness is to be assured into future, in days from current time. "
```

Comment:
shouldn't this be `in months from current time`?

Reply:
Yes. Pasto. Thanks. :-)